lib/matplotlib/axes/_axes.py

```diff
@@ -4281,19 +4281,26 @@ def invalid_shape_exception(csize, xsize):
             except ValueError:
                 pass  # Failed to convert to float array; must be color specs.
             else:
+                # handle the documented special case of a 2D array with 1
+                # row which as RGB(A) to broadcast.
+                if c.shape == (1, 4) or c.shape == (1, 3):
```

Comment:
OK, but do we want a 2-d array or a list of arrays as the special case?

Reply:
I don't understand.  Would you elaborate, please?  Why would there be a distinction here?

Reply:
The particular value for c is something like: `c = [[0.1, 0.1, 0.4]]`?

Reply:
We do `c = np.asanyarray(c, dtype=float)`  a few lines above to normalize `c`.  I do not think we want to get into the business of distinguishing between `[[a, b, c]]`, `[np.array([a, b, c])`, and `np.array([[a, b, c]])` as input.

Reply:
OK, but an unsqueezed 1x3 ndarray is a perfectly valid thing to pass for `c` if I have 3 data points.  So how can we tell them apart?

Reply:
Before it is squeezed, the dimensions are checked so that a 1x3 or 1x4 can be interpreted as a color.  The (3,) or (4,) 1-d cases, when the size matches the number of points, is interpreted as a sequence of values to be mapped. This is the documented way the ambiguity is resolved, based on a decision made long ago.  I just fouled it up in the last change I made to the code.  Thomas is cleaning up my mess.